File: 452/MinimumNumberOfArrowsToBurstBalloons.java
/*
https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/?envType=daily-question&envId=2024-03-18
*/

class CustomComparator implements Comparator<int[]>{
    @Override
    public int compare(int[] arr1, int[] arr2){
        return Integer.compare(arr1[1], arr2[1]);
    }
}

class Solution {
    public int findMinArrowShots(int[][] points) {
        if(points.length==0) return 0;
        Arrays.sort(points, new CustomComparator());
        int arrows=1, curend=points[0][1], sz=points.length;
        for(int i=1; i<sz; ++i){
            if(points[i][0]>curend){
                arrows++;
                curend=points[i][1];
            }
        }
        return arrows;
    }
}

File: 452/MinimumNumberOfArrowsToBurstBalloons.py
# https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/?envType=daily-question&envId=2024-03-18

class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:

        def custom_comparator(points):
            return points[1]

        if len(points)==0:
            return 0
        sorted_points=sorted(points, key=custom_comparator)
        arrows=1
        curend=sorted_points[0][1]
        sz=len(sorted_points)
        for i in range(1, sz, 1):
            if sorted_points[i][0]>curend:
                arrows+=1
                curend=sorted_points[i][1]
        return arrows